controllers/profiles/getOtherProfileByUsername.js

```diff
@@ -47,10 +64,12 @@ module.exports = async (req, res) => {
     copyUser.following_symbol = checkMoreOrLess(getFollowingCountResult);
     copyUser.follower_symbol = checkMoreOrLess(getFollowerCountResult);
 
-    copyUser.is_following = isFollowingResult;
-    if (copyUser.only_received_dm_from_user_following)
-      copyUser.isSignedMessageEnabled = isFollowingResult;
-    else copyUser.isSignedMessageEnabled = true;
+    if (req?.userId) {
+      copyUser.is_following = isFollowingResult;
+      if (copyUser.only_received_dm_from_user_following)
+        copyUser.isSignedMessageEnabled = isFollowingResult;
+      else copyUser.isSignedMessageEnabled = true;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The conditional check for `req?.userId` before setting `copyUser.is_following` and `copyUser.isSignedMessageEnabled` is a good practice as it ensures these properties are only set when `userId` is available, preventing potential null reference errors.
```diff
-    copyUser.is_following = isFollowingResult;
-    if (copyUser.only_received_dm_from_user_following)
-      copyUser.isSignedMessageEnabled = isFollowingResult;
-    else copyUser.isSignedMessageEnabled = true;
+    if (req?.userId) {
+      copyUser.is_following = isFollowingResult;
+      if (copyUser.only_received_dm_from_user_following)
+        copyUser.isSignedMessageEnabled = isFollowingResult;
+      else copyUser.isSignedMessageEnabled = true;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->